spec.md

```diff
@@ -0,0 +1,948 @@
+The value of this header is compared case-insensitively.
+
+> [!CAUTION]
+>
+> The exact semantics of the `CALCMEDLEY` header have not been defined yet.
+
+### 3.17. The `TITLE` Header
+
+```
+Required:     Yes
+Multi-Valued: No
+Since:        0.1.0
+```
+
+The `TITLE` header contains the title of the song.
+
+### 3.18. The `ARTIST` Header
+
+```
+Required:     Yes
+Multi-Valued: No
+Since:        0.1.0
+```
+
+The `ARTIST` header contains the artist of the song.
+
+### 3.19. The `YEAR` Header
+
+```
+Required:     No
+Multi-Valued: No
+Syntax:       4DIGIT
+Since:        0.2.0
+```
+
+The `YEAR` indicates the year in which the song was released.
+The value must be a positive integer.
+
+### 3.20. The `GENRE` Header
+
+```
+Required:     No
+Multi-Valued: Yes
+Since:        0.2.0
+```
+
+The `GENRE` defines the genre(s) of the song.
+Individual genre values MUST be compared case-insensitively.
+For consistency, it is usually best to capitalize genres.
+
+> [!CAUTION]
+>
+> Whether genres should be compared case-insensitively or not hasn't been decided yet.
+
+### 3.21. The `LANGUAGE` Header
+
+```
+Required:     No
+Multi-Valued: Yes
+Since:        0.2.0
+```
+
+The `LANGUAGE` header indicates the spoken or sung language(s) of a song.
+
+### 3.22. The `EDITION` Header
+
+```
+Required:     No
+Multi-Valued: Yes
+Since:        0.2.0
+```
+
+The `EDITION` indicates what edition of games a song belongs to.
+While this header is intended to hold commercially available editions (e.g. SingStar Pop Hits, GuitarHero Live) implementations MUST NOT reject a file based on the value of this header.
+A list of SingStar editions is available [here](https://github.com/bohning/usdb_syncer/wiki/SingStar-Editions).
+For arbitrary keywords see the `TAGS` header.
+
+### 3.23. The `TAGS` Header
+
+```
+Required:     No
+Multi-Valued: Yes
+Since:        1.1.0
+```
+
+The `TAGS` allow association of any reasonable keyword with a song.
+Implementations SHOULD compare tags in a case-insensitive manner.
+
+> [!CAUTION]
+>
+> Whether tags should be compared case-insensitively or not hasn't been decided yet.
+
+### 3.24. The `P1`, `P2`, … Headers
+
+```
+Required:     No
+Multi-Valued: No
+Since:        0.2.0
+```
+
+The headers `P1`, `P2`, … indicate the names of the voices of a song.
+These names correspond to the voices indicated by the `P1`, `P2`, … voice changes (see [section 3.3](#33-voice-changes)).
+If the voices correspond to different singers in the original song, the header values often indicate the names of the original singers.
+
+The association of header values to voices is defined by the numerical value after each `P` respectively,
+i.e. the header `P2` indicates the name of the voice whose notes are introduced by the `P2` voice change.
+Leading zeroes are insignificant, i.e. the headers `P1` and `P001` both refer to the same voice.
```

Comment:
Not sure if we should allow leading zeros

also zeroes vs zeros i'm not sure if we prefer brittish english or american english within this spec.

Reply:
In my current understanding the number of voices is basically unlimited. This comes with some challenges:
- How are leading zeros interpreted (Is `P010` the same as `P10`)?
- Implementations would have to protect themselves against malicious files where a `P10000000` player change/header is present. If implementations would naively allocate space for 10000000 voices, this could easily cause crashes.

But I think it would make sense, at least syntactically, to just allow at most 9 voices (`P1` thru `P9`). That would solve these challenges. I'm gonna change this and maybe we can revisit this at some point in the future.

Reply:
And we should absolutely be consistent with American vs. British English. I think I mostly used American English (e.g. "standardized") because most other specs seem to use that. But I'm open to change that.